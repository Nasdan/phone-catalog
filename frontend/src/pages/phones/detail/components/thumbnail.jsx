import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import { imageContainer } from '../../../../common/styles/image';

export const Thumbnail = ({ imageUrl }) => (
  <Card>
    <CardContent>
      <div className={imageContainer}>
        <img src={imageUrl} />
      </div>
    </CardContent>
  </Card>
);

Thumbnail.propTypes = {
  imageUrl: PropTypes.string.isRequired,
};
