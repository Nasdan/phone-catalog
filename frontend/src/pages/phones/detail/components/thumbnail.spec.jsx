import React from 'react';
import { shallow } from 'enzyme';
import { Thumbnail } from './thumbnail';

describe('pages/phones/detail/components/thumbnail specs', () => {
  it('should render as expected when feeding imageUrl', () => {
    // Arrange
    const imageUrl = 'test imageUrl';

    // Act
    const component = shallow(<Thumbnail imageUrl={imageUrl} />);

    // Assert
    expect(component).toMatchSnapshot();
  });
});
