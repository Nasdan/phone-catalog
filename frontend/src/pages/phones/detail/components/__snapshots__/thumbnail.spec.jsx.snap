// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/phones/detail/components/thumbnail specs should render as expected when feeding imageUrl 1`] = `
<WithStyles(Card)>
  <WithStyles(CardContent)>
    <div
      className="css-1fv1iua"
    >
      <img
        src="test imageUrl"
      />
    </div>
  </WithStyles(CardContent)>
</WithStyles(Card)>
`;
